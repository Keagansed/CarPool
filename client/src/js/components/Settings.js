import React, { Component } from 'react';
import { observer } from "mobx-react";
import LoginStore from '../stores/LoginStore';
import { Link } from 'react-router-dom';

import Navbar from './Navbar'
// import 'bootstrap/dist/css/bootstrap.min.css';



@observer class Settings  extends Component {

    handleLogout = () =>
    {
        LoginStore.logOut();
    }

    render(){
        return(
            <div>
                <h1>Settings</h1>
                <Link to={"/"} onClick={this.handleLogout}>Log Out</Link> 
                <Navbar token={ this.props.store.token }/>
            </div>
        );
    }
}

export default Settings;