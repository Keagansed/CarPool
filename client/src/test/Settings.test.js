import React from 'react';
import Settings from '../js/components/Settings';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'

describe('Settings Page Component', () => {
    const mockStore = configureStore()
    let store, container

    beforeEach(() => {
        store = mockStore();
        container = shallow(<Settings store={store} />)
    });

    it('renders correctly', () => {
        expect(container.length).toEqual(1)
    });
});