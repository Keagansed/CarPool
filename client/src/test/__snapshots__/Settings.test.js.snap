// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Page Component captures snapshot 1`] = `
<div>
  <h1>
    Settings
  </h1>
  <a
    href="/"
    onClick={[Function]}
  >
    Log Out
  </a>
  <div
    className="navbar-component navbarBottom"
  >
    <div
      className="container-fluid p-0"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12 "
        >
          <ul
            className="nav nav-pills navbarBottom text-center"
          >
            <li
              className="nav-item ml-auto w-25"
            >
              <a
                className="nav-link no-hover"
                href="/Setting"
                onClick={[Function]}
              >
                <i
                  className="fa fa-cog fa-2x text-secondary d-inline"
                />
                Â 
              </a>
            </li>
            <li
              className="nav-item w-25"
            >
              <a
                className="nav-link no-hover"
                href="/HomePage"
                onClick={[Function]}
              >
                <i
                  className="fa fa-fw fa-car fa-2x d-inline text-secondary"
                />
              </a>
            </li>
            <li
              className="nav-item mr-auto text-secondary w-25"
            >
              <a
                className="nav-link no-hover"
                href="/ProfilePage/undefined"
                onClick={[Function]}
              >
                <i
                  className="fa fa-fw fa-2x fa-user d-inline text-secondary"
                />
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
